ui/page/wallets/wallet_page.go

```diff
@@ -979,4 +987,32 @@ func (pg *WalletPage) HandleUserInteractions() {
 // Part of the load.Page interface.
 func (pg *WalletPage) OnNavigatedFrom() {
 	pg.closePopups()
+	pg.ctxCancel()
+}
+
+// listenForsyncNotifications spuns a go-routine that listens for
+// Wallet notifications. Currently we are interested in reading from
+// NotificationsUpdate channel, to prevent the channel from getting
+// filled thereby blocking Wallet create/delete actions.
```

Comment:
Seems to me that this is a bad design. Why do we have this channel in the first place and why do we need to drain it this way to prevent it from blocking?

Reply:
The channel provides sync status updates mainly used by the overview page. If it navigates away from the overview page the channel is no longer consumed, the channel gets filled up and block.

Reply:
It'd be better for pages that require such updates to create the channel and close it when done with it. A universal channel that is not always used and could become blocked isn't ideal. I'll propose some changes in a moment.

Reply:
> It'd be better for pages that require such updates to create the channel and close it when done with it. A universal channel that is not always used and could become blocked isn't ideal. I'll propose some changes in a moment.

I had thought about that, but a single universal channel is used to deliver tx/mixer/politeia/sync/rescan notifications. Changes will require breaking this up into units, so pages can create the units they need and close them on exit. This changes the scope of the task from fixing the bugs mentioned in the PR to refactoring the listeners. Pointing that out before I proceed.